### PR TITLE
Dependencies: Update ocaml-h2 and piaf to latest versions

### PR DESCRIPTION
**Pull request by lessp**
I was hitting errors which were fixed in: https://github.com/anmonteiro/ocaml-h2/pull/116

So thought I'd create a PR pulling in those changes. Let me know if there's anything else that needs to be done for this to go in! 🙂

### CODE REVIEW

File: h2.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@reason-native-web/h2",
-  "version": "0.5.1001",
+  "version": "0.5.1002",
```
Comment (ulrikstrid) [inline h2.json]: 0.6.0, denna är släppt på opam också

File: piaf.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@reason-native-web/piaf",
-  "version": "1.0.2000",
+  "version": "1.0.3000",
```
Comment (ulrikstrid) [inline piaf.json]: Hasse tänkt mig 1.1.0, detta är okej också iofs

**Review by ulrikstrid — CHANGES_REQUESTED** [review]

**Comment by ulrikstrid** [comment]
You need to update Piaf's dependencies as well to pick up httpaf (h1) and h2

**Comment by lessp** [comment]
Would probably also have to push a release of `h2` before bumping `piaf`?

**EDIT:** ...or, looks like CI publishes releases, so guess it'd be enough to match the coming version number?

**Comment by anmonteiro** [comment]
you need to mirror the new dependency on `gluten-lwt{-unix}` too

**Comment by anmonteiro** [comment]
Please add the remaining dependencies on `gluten-lwt` and `gluten-lwt-unix` as specified in https://github.com/anmonteiro/ocaml-h2/blob/0.6.0/h2-lwt.opam and https://github.com/anmonteiro/ocaml-h2/blob/0.6.0/h2-lwt-unix.opam

**Comment by lessp** [comment]
> Please add the remaining dependencies on `gluten-lwt` and `gluten-lwt-unix` as specified in https://github.com/anmonteiro/ocaml-h2/blob/0.6.0/h2-lwt.opam and https://github.com/anmonteiro/ocaml-h2/blob/0.6.0/h2-lwt-unix.opam

Thanks, missed those. 🙂

File: h2-lwt-unix.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@reason-native-web/h2-lwt-unix",
-  "version": "0.5.1001",
+  "version": "0.5.1002",
```
Comment (ulrikstrid) [inline h2-lwt-unix.json]:
```suggestion
  "version": "0.6.0",
```

File: h2-lwt-unix.json [hunk 4]
```diff
@@ -12,7 +12,8 @@
   "dependencies": {
     "ocaml": ">=4.6.0",
     "@opam/faraday-lwt-unix": "*",
-    "@reason-native-web/h2-lwt": ">=0.5.0",
+    "@opam/gluten-lwt-unix": "*",
+    "@reason-native-web/h2-lwt": "^0.5.1001",
```
Comment (ulrikstrid) [inline h2-lwt-unix.json]:
```suggestion
    "@reason-native-web/h2-lwt": "^0.6.0",
```

File: h2-lwt.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@reason-native-web/h2-lwt",
-  "version": "0.5.1001",
+  "version": "0.5.1002",
```
Comment (ulrikstrid) [inline h2-lwt.json]:
```suggestion
  "version": "0.6.0",
```

File: h2-lwt.json [hunk 6]
```diff
@@ -11,8 +11,9 @@
   "files": ["ocaml-h2"],
   "dependencies": {
     "ocaml": ">=4.6.0",
-    "@reason-native-web/h2": ">=0.5.0",
+    "@reason-native-web/h2": "^0.5.1001",
```
Comment (ulrikstrid) [inline h2-lwt.json]:
```suggestion
    "@reason-native-web/h2": "^0.6.0",
```

File: piaf.json (hunk 2 above)
Comment (ulrikstrid) [inline piaf.json]:
```suggestion
  "version": "1.1.0",
```

File: piaf.json [hunk 7]
```diff
@@ -15,11 +15,12 @@
     "@opam/ocaml-syntax-shims": "*",
     "@opam/bigstringaf": ">=0.5.0",
     "@reason-native-web/h1-lwt-unix": "~1.1.0",
-    "@reason-native-web/h2-lwt-unix": "~0.5.1001",
+    "@reason-native-web/h2-lwt-unix": "^0.5.1002",
```
Comment (ulrikstrid) [inline piaf.json]:
```suggestion
    "@reason-native-web/h2-lwt-unix": "^0.6.0",
```

**Review by ulrikstrid — CHANGES_REQUESTED** [review]

File: h2-lwt-unix.json [hunk 8]
```diff
@@ -12,7 +12,8 @@
   "dependencies": {
     "ocaml": ">=4.6.0",
     "@opam/faraday-lwt-unix": "*",
-    "@reason-native-web/h2-lwt": ">=0.5.0",
+    "@opam/gluten-lwt-unix": "*",
```
Comment (anmonteiro) [inline h2-lwt-unix.json]:
```suggestion
    "@opam/gluten-lwt-unix": ">=0.2.0",
```

File: h2-lwt.json [hunk 9]
```diff
@@ -11,8 +11,9 @@
   "files": ["ocaml-h2"],
   "dependencies": {
     "ocaml": ">=4.6.0",
-    "@reason-native-web/h2": ">=0.5.0",
+    "@reason-native-web/h2": "^0.6.0",
     "@opam/dune": ">=1.5.0",
-    "@opam/lwt": "*"
+    "@opam/lwt": "*",
+    "@opam/gluten-lwt": "*"
```
Comment (anmonteiro) [inline h2-lwt.json]:
```suggestion
    "@opam/gluten-lwt": ">=0.2.0"
```

File: piaf.json [hunk 10]
```diff
@@ -15,11 +15,12 @@
...
     "@opam/logs": "*",
     "@opam/lwt": "*",
-    "@reason-native-web/lwt_ssl": ">=1.1.3000",
+    "@reason-native-web/lwt_ssl": "^1.1.3000",
     "@opam/uri": "*",
-    "@opam/magic-mime": "*"
+    "@opam/magic-mime": "*",
+    "@opam/gluten-lwt-unix": "*"
```
Comment (anmonteiro) [inline piaf.json]:
```suggestion
    "@opam/gluten-lwt-unix": ">=0.2.0"
```

**Comment by anmonteiro** [comment]
this still needs to pull "h1" from my httpaf repo, it runs on the gluten runtime too, and Piaf expects that.

**Comment by ulrikstrid** [comment]
I opened #12 and added you as a co-author @lessp. Thanks for this work but I'll close in favor of the other PR since it's a bit hairy with all the versions 😅

**Pull request closed without merging**